lib/src/arithmetic/fixed_point/fixed_point_logic.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// fixed_point_logic.dart
+// Implementation of Fixed Point objects
+//
+// 2024 September 25
+// Author: Soner Yaldiz <soner.yaldiz@intel.com>
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/src/exceptions.dart';
+
+/// Fixed point logic representation
+class FixedPoint extends Logic {
+  /// [integerWidth] is the width of bits reserved for integer part
+  late final int integerWidth;
+
+  /// [fractionWidth] is the width of bits reserved for fractional part
+  late final int fractionWidth;
+
+  static int _fixedPointWidth(int a, int b) => 1 + a + b;
+
+  /// [FixedPoint] Constructor
+  FixedPoint(
+      {required this.integerWidth, required this.fractionWidth, super.name})
```

Comment:
might be nice to include `super.naming` as well?

Reply:
What do you mean given that there is an optional `super.name` argument?

Reply:
Is there an example?

Reply:
Here's some code (and doc comments) as it relates to naming:

https://github.com/intel/rohd/blob/3e2d24a78adc7579fb7d4a99452d2105a2e70b65/lib/src/signals/logic.dart#L231-L246

https://github.com/intel/rohd/blob/main/lib/src/utilities/naming.dart

does that help?